app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -202,45 +205,45 @@ private HashMap<Integer, ParsePolygon> getActualGridSquare() {
         HashMap<Integer, ArrayList<ParseGeoPoint>> grid = new HashMap<>();
         //squareSize is a "debug number and it will change depends of the speedVector"
         double squareSize = 0.005;
-        int numSquareSizeGrid = 3;
-        double cornerLatitudeCC = currentLocation.getLatitude() - positiveRemainder(squareSize, currentLocation.getLatitude());
-        double cornerLongitudeCC = currentLocation.getLongitude() - positiveRemainder(squareSize, currentLocation.getLongitude());
+        // abbr. CS means central square
+        double cornerLatitudeCS = currentLocation.getLatitude() - positiveRemainder(squareSize, currentLocation.getLatitude());
+        double cornerLongitudeCS = currentLocation.getLongitude() - positiveRemainder(squareSize, currentLocation.getLongitude());
         //saving the number of points in the grid
-        LatLng[][] gridCorners = new LatLng[numSquareSizeGrid + 1][numSquareSizeGrid + 1];
+        LatLng[][] gridCorners = new LatLng[SQUARE_GRID_LENGTH + 1][SQUARE_GRID_LENGTH + 1];
         for (int i = 0; i < gridCorners.length; i++) {
             for (int j = 0; j < gridCorners[0].length; j++) {
                 gridCorners[i][j] = new LatLng(
-                        cornerLatitudeCC - (1 - i) * squareSize,
-                        cornerLongitudeCC - (1 - j) * squareSize
+                        cornerLatitudeCS - (1 - i) * squareSize,
+                        cornerLongitudeCS - (1 - j) * squareSize
                 );
             }
         }
-        //k is nine because the grid has 9 squares
-        for (int k = 0; k < 9; k++) {
-            grid.put(k, new ArrayList<>());
+        for (int keySquare = 0; keySquare < SQUARE_GRID_3X3_COUNT; keySquare++) {
```

Comment:
I'd also probably add a comment describing what this block is attempting to do since I'm still a little unsure on that.

// This block aims to calculate and store X, returning a map of <blank> to <blank> which is used to do <blah>
// We start by examining each square in our grid, selecting it's column and <blah blah blah>...